### PR TITLE
feat(backend): exposed a method to query token holders of bridge contracts

### PR DESCRIPTION
**Pull request by icerove**
the query is for bridge tracking :
for each token, who is mint and hold this token

### CODE REVIEW

File: backend/src/db-utils.js [hunk 1]
```diff
@@ -528,6 +528,24 @@ const queryPartnerUniqueUserAmount = async () => {
   );
 };
 
+// bridge query
+const queryBridgeUser = async (accountId) => {
```
Comment (frol) [inline backend/src/db-utils.js:532]: You lost me here with the naming. To query a bridge user you need to do `SELECT * FROM accounts WHERE account_id = ?`. The implementation clearly does not match it.

File: backend/src/db-utils.js [hunk 2]
```diff
@@ -528,6 +528,24 @@ const queryPartnerUniqueUserAmount = async () => {
...
+  return await queryRows(
+    [
+      `select 
+        distinct transactions.signer_account_id as holder
+      from receipts 
+      join action_receipt_actions on receipts.receipt_id = action_receipt_actions.receipt_id  
+      join transactions on transactions.transaction_hash = receipts.originated_from_transaction_hash 
+      where receipts.receiver_account_id = :account_id
+      and action_receipt_actions.action_kind = 'FUNCTION_CALL' 
+      and action_receipt_actions.args->> 'method_name' = 'mint'`,
```
Comment (frol) [inline backend/src/db-utils.js:542]: 1. You probably only want successful receipts
2. nit: Please use upper-case for the SQL keywords (follow the code style)

**Review by frol — CHANGES_REQUESTED** [review]

File: backend/src/db-utils.js (hunk 1 above)
Comment (icerove) [reply backend/src/db-utils.js:532]: change to `queryBridgeTokenHolder`

File: backend/src/db-utils.js (hunk 2 above)
Comment (icerove) [reply backend/src/db-utils.js:542]: 1. do not need that since query also needs failure one
2. fix the case pro

File: backend/src/db-utils.js [hunk 3]
```diff
@@ -528,6 +528,24 @@ const queryPartnerUniqueUserAmount = async () => {
   );
 };
 
+// bridge query
+const queryBridgeTokenHolder = async (accountId) => {
```
Comment (frol) [inline backend/src/db-utils.js:532]: You seem to want to have a list of accounts (holders) which minted tokens through bridge, right? If that is the case:

1. the name has to be plural (`queryBridgeTokenHolders`)
2. `accountId` in the arguments misleads a reader; I suggest `bridgeTokenContractId`

File: backend/src/wamp.js [hunk 4]
```diff
@@ -305,6 +305,11 @@ wampHandlers["partner-unique-user-amount"] = async () => {
   return await stats.getPartnerUniqueUserAmount();
 };
 
+// bridge query
+wampHandlers["near-bridge-user"] = async ([accountId]) => {
```
Comment (frol) [inline backend/src/wamp.js:309]:
```suggestion
wampHandlers["get-bridge-token-holders"] = async ([accountId]) => {
```

**Review by frol — APPROVED** [review]

**Review by frol — APPROVED** [review]

**Comment by chefsale** [comment]
@icerove Resolved the fossa key issue.

**Pull request merged**